src/main/protobuf/soc/moves.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto2";
+
+package protos.soc;
+
+import "soc/inventory.proto";
+import "soc/board.proto";
+import "soc/state.proto";
+
+enum ActionType {
+    ROLL_DICE = 0;
+    END_TURN = 1;
+    INITIAL_PLACEMENT = 2;
+    DISCARD_RESOURCES = 3;
+    MOVE_ROBBER_AND_STEAL = 4;
+    BUY_DEVELOPMENT_CARD = 5;
+    BUILD_ROAD = 6;
+    BUILD_SETTLEMENT = 7;
+    BUILD_CITY = 8;
+    PORT_TRADE = 9;
+    PLAY_KNIGHT = 10;
+    PLAY_YEAR_OF_PLENTY = 11;
+    PLAY_MONOPOLY = 12;
+    PLAY_ROAD_BUILDER = 13;
+}
+
+message HiddenCard {
+    repeated Player viewable_by = 1;
+    oneof card {
+        Resource resource_card = 2;
+        DevelopmentCard development_card = 3;
+    }
+}
+
+message ActionSpecification {
+    required ActionType type = 1;
+    oneof payload {
+        InitialPlacement initial_placement_payload = 2;
+        DiscardResources discard_resources_payload = 3;
+        MoveRobberAndSteal move_robber_and_steal_payload = 4;
+        BuildRoad build_road_payload = 5;
+        BuildSettlement build_settlement_payload = 6;
+        BuildCity build_city_payload = 7;
+        PortTrade port_trade_payload = 8;
+        MoveRobberAndSteal play_knight_payload = 9;
+        PlayYearOfPlenty play_year_of_plenty_payload = 10;
+        PlayMonopoly play_monopoly_payload = 11;
+        PlayRoadBuilder play_road_builder_payload = 12;
+    }
+}
+
+message ActionResult {
+    required ActionType type = 1;
+    oneof payload {
```

Comment:
```
oneof payload {
        InitialPlacement initial_placement_payload = 2;
        DiscardResources discard_resources_payload = 3;
        MoveRobberAndSteal move_robber_and_steal_payload = 4;
        BuildRoad build_road_payload = 5;
        BuildSettlement build_settlement_payload = 6;
        BuildCity build_city_payload = 7;
        PortTrade port_trade_payload = 8;
        MoveRobberAndSteal play_knight_payload = 9;
        PlayYearOfPlenty play_year_of_plenty_payload = 10;
        PlayMonopoly play_monopoly_payload = 11;
        PlayRoadBuilder play_road_builder_payload = 12;
    }
```

Reply:
For example:
```
message GameEvent {
  int32 position = 1; 
  GameAction action = 2;
  oneof payload {
        InitialPlacement initial_placement_payload = 2;
        DiscardResources discard_resources_payload = 3;
        MoveRobberAndSteal move_robber_and_steal_payload = 4;
        BuildRoad build_road_payload = 5;
        BuildSettlement build_settlement_payload = 6;
        BuildCity build_city_payload = 7;
        PortTrade port_trade_payload = 8;
        MoveRobberAndSteal play_knight_payload = 9;
        PlayYearOfPlenty play_year_of_plenty_payload = 10;
        PlayMonopoly play_monopoly_payload = 11;
        PlayRoadBuilder play_road_builder_payload = 12;
    }
  ActionResult result = 4;
}
```

```
message ActionResult {
  int32 roll = 1;
  HiddenCard card = 2;
  map<int32, ResourceTransaction> resources_transacted = 3; // Only shows publically transacted resources
}
```

Reply:
What is this line  in GameEvent supposed to mean? 
`  GameAction action = 2;`

Reply:
Sorry, I was just pasting that in quickly because I was in a rush `GameAction = ActionType`. So within the context of this PR, it would be:
```
message GameEvent {
  Player player = 1;
  ActionType action = 2;
  oneof payload {
        ....
    }
  ActionResult result = 4;
}
```

That said I think it would be better if a `GameEvent` just returned an action specification as a subfield.
So:

```
message GameEvent {
  required Player player = 1;
  required ActionSpecification = 2;
  required ActionResult result = 3;
}
```

And `ActionSpecification` how you defined it. A `GameAction` and an optional (typed) specification if the action has one. SO if the `GameAction` is `ROLL` then there would be no specification.

Reply:
I dont think we need `player` to be specified in this since this is just a proto to define GameActions. Another proto in `SOC-Server` which goes between client and server should specify player with saying what the ActionEvent is.

Reply:
1) I really think all the proto files should all live in the same place

2) I think we are saying the same thing about wrapping a `GameEvent` in an `ActionEvent`. I just have been using this terminology:
```
message GameEvent {
  required Player player = 1;
  required ActionSpecification = 2;
  optional ActionResult result = 3;
}
```

Could you incorporate that in this PR?

An ActionSpecification is simply:

```
message ActionSpecification
    required GameAction type = 1;
    oneof specification {
        InitialPlacement initial_placement_payload = 2;
        DiscardResources discard_resources_payload = 3;
        MoveRobberAndSteal move_robber_and_steal_payload = 4;
        BuildRoad build_road_payload = 5;
        BuildSettlement build_settlement_payload = 6;
        BuildCity build_city_payload = 7;
        PortTrade port_trade_payload = 8;
        MoveRobberAndSteal play_knight_payload = 9;
        PlayYearOfPlenty play_year_of_plenty_payload = 10;
        PlayMonopoly play_monopoly_payload = 11;
        PlayRoadBuilder play_road_builder_payload = 12;
    }
```

Then there is an rpc call called `makeMove(actionSpecification, player)`

Could you incorporate that pattern in this PR?